src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs

```diff
@@ -331,7 +331,7 @@ private void WriteIPv6Bytes(Span<byte> destination)
         private void WriteIPv4Bytes(Span<byte> destination)
         {
             uint address = PrivateAddress;
-            MemoryMarshal.Write(destination, ref address);
+            MemoryMarshal.Write(destination, in address);
```

Comment:
Should we change this to:
```C#
MemoryMarshal.Write(destination, PrivateAddress);
```
? Seems like the primary motivation behind changing the parameter from ref to in is so that it can use rvalues

Reply:
We definitely can. I had just gone with the most straightforward set of changes to start since we're trying to land this in .NET 8